compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp

```diff
@@ -790,6 +791,18 @@ FailureOr<Operation *> hoistOutOfDispatch(RewriterBase &rewriter,
 // Utilities to make a dispatch region isolated from above
 //===---------------------------------------------------------------------===//
 
+static bool isAttentionMaskGenerator(Operation *op) {
+  for (OpOperand &use : op->getUses()) {
```

Comment:
Shouldn't we check that all uses are mask?

Reply:
no, as long as even one producer is a mask, we should always clone. The mask is too big to be ever materialized.

Reply:
ok, the logic is a bit convoluted. If there is a usage that is not a mask, then it will get materialized anyway... So realistically you are looking for when all uses are the mask.